spring-cloud-stream-integration-tests/src/test/java/org/springframework/cloud/stream/config/aggregate/AggregateApplicationTests.java

```diff
@@ -41,25 +42,28 @@
  * @author Oleg Zhurakousky
  */
 @RunWith(SpringJUnit4ClassRunner.class)
+@DirtiesContext
 public class AggregateApplicationTests {
 
-	@Before
-	public void before() {
-		System.setProperty("server.port", "0");
-		System.setProperty("spring.main.allow-bean-definition-overriding", "true");
-	}
-
 	@Test
 	@SuppressWarnings("unchecked")
 	public void testAggregateApplication() throws Exception {
 		ConfigurableApplicationContext context = new AggregateApplicationBuilder(
-				TestSupportBinderAutoConfiguration.class).web(false).from(TestSource.class).to(TestProcessor.class).run();
+				FooConfig.class).web(false).from(TestSource.class).to(TestProcessor.class).run();
 		TestSupportBinder testSupportBinder = (TestSupportBinder) context.getBean(BinderFactory.class).getBinder(null,
 				MessageChannel.class);
 		MessageChannel processorOutput = testSupportBinder.getChannelForName("output");
 		Message<String> received = (Message<String>) (testSupportBinder.messageCollector().forChannel(processorOutput)
 				.poll(5, TimeUnit.SECONDS));
 		Assert.assertThat(received, notNullValue());
 		Assert.assertTrue(received.getPayload().endsWith("processed"));
+
+		context.close();
+	}
+
+	@Configuration
+	@EnableAutoConfiguration
+	static class FooConfig{
```

Comment:
I believe this class should have a much proper name. And we need to have in the commit message a point "why?" - the `TestSupportBinderAutoConfiguration` is really an auto-configuration so we need to rely on the Boot in this case to pick up everything properly from the classpath and resources.

Reply:
True, the name is mea meaningless, but given that we're already "soft-deprecated" aggregate applications (by simply removing documentation for it), we can live with it until we remove it.